diff --git a/qtbase/mkspecs/linux-aarch64-gnu-g++/qmake.conf b/qtbase/mkspecs/linux-aarch64-gnu-g++/qmake.conf
index 2dc57b09fa..90dce0b081 100644
--- a/qtbase/mkspecs/linux-aarch64-gnu-g++/qmake.conf
+++ b/qtbase/mkspecs/linux-aarch64-gnu-g++/qmake.conf
@@ -10,15 +10,27 @@ include(../common/linux.conf)
 include(../common/gcc-base-unix.conf)
 include(../common/g++-unix.conf)
 
+QMAKE_INCDIR_OPENGL     = $$(OPENGL_INC_DIR)
+QMAKE_LIBDIR_OPENGL     = $$(OPENGL_LIB_DIR)
+
+QMAKE_INCDIR_OPENGL_ES2 = $(OPENGL_INC_DIR)
+QMAKE_LIBDIR_OPENGL_ES2 = $$(OPENGL_LIB_DIR)
+QMAKE_INCDIR_EGL        = $$(OPENGL_INC_DIR)
+QMAKE_LIBDIR_EGL        = $$(OPENGL_LIB_DIR)
+QMAKE_LIBS_OPENGL_ES2   = $$(OPENGL_LIB_DIR)
+QMAKE_LIBS_EGL          = $$(OPENGL_LIB_DIR)
+
+QMAKE_LFLAGS            += -Wl,--rpath-link,$$QMAKE_LIBDIR_OPENGL
+
 # modifications to g++.conf
-QMAKE_CC                = aarch64-linux-gnu-gcc
-QMAKE_CXX               = aarch64-linux-gnu-g++
-QMAKE_LINK              = aarch64-linux-gnu-g++
-QMAKE_LINK_SHLIB        = aarch64-linux-gnu-g++
+QMAKE_CC                = $$(CROSSTOOL_PREFIX)gcc
+QMAKE_CXX               = $$(CROSSTOOL_PREFIX)g++
+QMAKE_LINK              = $$(CROSSTOOL_PREFIX)g++
+QMAKE_LINK_SHLIB        = $$(CROSSTOOL_PREFIX)g++
 
 # modifications to linux.conf
-QMAKE_AR                = aarch64-linux-gnu-ar cqs
-QMAKE_OBJCOPY           = aarch64-linux-gnu-objcopy
-QMAKE_NM                = aarch64-linux-gnu-nm -P
-QMAKE_STRIP             = aarch64-linux-gnu-strip
+QMAKE_AR                = $$(CROSSTOOL_PREFIX)ar cqs
+QMAKE_OBJCOPY           = $$(CROSSTOOL_PREFIX)objcopy
+QMAKE_NM                = $$(CROSSTOOL_PREFIX)nm -P
+QMAKE_STRIP             = $$(CROSSTOOL_PREFIX)strip
 load(qt_config)
